uber/site_sections/registration.py

```diff
@@ -231,11 +242,18 @@ def check_in(self, session, id, badge_num, age_group, message=''):
                 maybe_dupe = session.query(Attendee).filter_by(badge_num=badge_num, badge_type=attendee.badge_type)
                 if maybe_dupe.count():
                     message = 'That badge number already belongs to ' + maybe_dupe.first().full_name
+
+            if group:
+                session.match_to_group(attendee, session.group(group))
```

Comment:
I added ``and not attendee.group`` to this line since otherwise it blew up on people who already had groups.